### ISSUE TITLE
[iOS] The turn instructions are white on white in CarPlay

### ISSUE THREAD

**Issue by rtsisyk** [organicmaps/organicmaps#6379]
100% reproducibility....

![IMG_2235 copy](https://github.com/organicmaps/organicmaps/assets/1799054/7c3379d6-b04b-4f9b-8850-8a3f294f2173)

**Comment by dvdmrtnz** [comment]
* Same as https://github.com/organicmaps/organicmaps/issues/2706

**Comment by kirylkaveryn** [comment]
So the bug was in CarPlayRouter:
  ```swift
private func createUpcomingManeuvers() -> [CPManeuver] {
     ...
    primaryManeuver.instructionVariants = [instructionVariant]
    if let imageName = routeInfo.turnImageName,
      let symbol = UIImage(named: imageName) {
      primaryManeuver.symbolSet = CPImageSet(lightContentImage: symbol,
                                             darkContentImage: symbol)
    }
```
Each of the Light and Dark modes should use separate images.
Smth like that:

 ```swift
private func createUpcomingManeuvers() -> [CPManeuver] {
    ...
    if let imageName = routeInfo.turnImageName,
      let symbol = UIImage(named: imageName) {
      var symbolForLightContent = symbol
      var symbolFoDarkContent = symbol
      
      if #available(iOS 13.0, *) {
        symbolForLightContent = symbol.withTintColor(.black, renderingMode: .alwaysTemplate)
        symbolFoDarkContent = symbol.withTintColor(.white, renderingMode: .alwaysTemplate)
      }
      
      primaryManeuver.symbolSet = CPImageSet(lightContentImage: symbolForLightContent,
                                             darkContentImage: symbolFoDarkContent)
      
    }
```

But... there is another small problem: CarPlay overrides a tint color of the maneuver's symbol and ignores black/white while using template rendering (if we use the default rendering, the symbol always be white - as of now). 

So there are 3 approaches:
1. use default system blue/lightblue tinting (which _imho_ looks very well and is consistent with other car play elements) - see screenshots (dark gray line with "next turn" seems like default behaviour):
<img width="802" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/7c279d56-1dba-432e-ad4a-5e4627d9c1ef">
<img width="801" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/bed6774c-0ba3-4f4e-a651-a419f2e31a75">

2. redraw UIImage with another tinting using UIGraphicsImageRenderer or UIGraphicsContext (needs to add extension to UIImage)
<img width="803" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/9ef9f24e-a395-4b0e-b3f4-53308b235ac5">

3. add for each image the black duplicate for dark content (bad one)

**Comment by biodranik** [comment]
Thanks for digging this up!

1. Someone says [There is no way of theming a CarPlay application as of iOS 15.4](https://stackoverflow.com/questions/71502975/tint-color-for-carplay-ui)
2. symbolSet is deprecated, can we use [symbolImage](https://developer.apple.com/documentation/carplay/cpmaneuver/3183006-symbolimage) instead? Will the system still override the tint color?
3. Does it make sense to reuse the same colors that are used in Apple Maps, or introduce our "firm" green background, similar to Google, to make the CarPlay experience more "familiar"?
4. Default system tinting looks good enough for a quick fix. But see below.
5. The second direction instruction looks bad in the light theme. The text "Then" should be white. The blue arrow is not very visible on the dark background. A light-blue version of that smaller direction arrow would be better, a white one would be even better.
6. Redrawing UIImages looks like an overkill, but if the resulting images are cached, and are redrawn/created only once, then it may be an acceptable workaround if nothing else works.
7. (A separate issue) There is too much free space on the left and right from the direction arrow. It makes sense to redesign it (if it's possible), by drawing the arrow on the left, and the distance on the right (does Apple Maps do that?). And free more space for the name of the next street, and for the next direction instruction.

**Comment by kirylkaveryn** [comment]
@biodranik 
1. Changing of theme/UI of CarPlay is really limited...
2. symbolImage also works well using template rendering with  `primaryManeuver.symbolImage = symbol.withRenderingMode(.alwaysTemplate)`. But the system overrides tint too... Every UIView subclass that has setted up with tint color will override UIImage color with  `.alwaysTemplate` 
3. This is a little bit tricky... Apple says about background color id CarPlay: `The system determines whether guidanceBackgroundColor meets contrast requirements, and uses the default color when the provided color doesn’t meet those requirements. The system adjusts font color to correspond with the guidance background color.`
But OM's "firm" `#188044` color doesn't meet such requirements... for both "white on green"  (almost Ok)and "black on green" (totally fails).
CarPlay detects it and overrides our color with the default white/darkGray backgrounds...
And Google's color `#2E6433` is fully meet for "white on green" so they use that combination for both light and dark themes.

What can we do? Slightly change our color so "white on green" will pass.
Our color:
<img width="733" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/bcde881e-89e1-4436-9147-913380aa4a8b">
<img width="733" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/f8dccb02-0492-41f6-93f5-59d0c1894db1">

Google:
**<img width="740" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/c588bd8b-3dd1-4d9a-a5fe-8bd06453ce30">**
<img width="803" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/7b398607-7da1-4af9-ba65-430dfc03455d">

4. Ok
5. It can be fixed also! But with the small trick also. If we override backgroundColor with our custom color it will be used for both light/dark themes. But  if we add the color into color assets we can setup **dynamically changing color** which automatically reacts on the theme changing without ANY additional handling. It is really easy to use.
<img width="1304" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/e4c34a00-d72b-442d-bb81-137a9816b728">
```swift
if #available(iOS 13.0, *) {
      mapTemplate.guidanceBackgroundColor = UIColor(named: "cp_background")
    }
```
<img width="798" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/75a2676d-f963-46fb-ab53-3724bda5e0d0">
6. I don't like thais approach too...
7. Arrangement, fonts, insets and spacings in CarPlay direction cannot be changed.

**Comment by biodranik** [comment]
Google has good contrast, which is very important. Can we use a similar approach or Apple Maps colors? Keeping a "firm" color is not as important for a separate carplay display as showing high-contrast directions for drivers.

Let's fix these issues in the way that you think is the best one.
